cpp/src/join/conditional_join.cu

```diff
@@ -348,14 +443,13 @@ std::unique_ptr<rmm::device_uvector<size_type>> conditional_left_semi_join(
   rmm::mr::device_memory_resource* mr)
```

Comment:
**☑️ todo:** ‏ Please use `rmm::device_async_resource_ref` (not pointer)

Reply:
See #15498 for details / examples. It should be a straightforward replacement.

Reply:
@DanialJavady96 I can launch another round of CI tests once this and the below (line 366) `mr` is migrated to the new async resource ref.